libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -216,6 +235,26 @@ void AC_Avoid::adjust_velocity(Vector3f &desired_vel_cms, bool &backing_up, floa
             }
         }
     }
+    // limit acceleration
+    limit_accel(desired_vel_cms_original, desired_vel_cms, dt);
+}
+
+// Limit acceleration so that change of velocity output by avoidance library is controlled
+void AC_Avoid::limit_accel(const Vector3f &original_vel, Vector3f &modified_vel, float dt)
+{
+    if (original_vel == modified_vel || is_zero(_accel_max) || is_zero(dt)) {
+        // we can't limit accel if any of these conditions are true
+        return;
+    }
+    // max accel in cm
+    const float max_accel_cm = _accel_max * 100.0f;
+    // accel desired by avoidance library
+    const Vector3f accel = (modified_vel - original_vel)/dt;
+    if (accel.length() > max_accel_cm) {
+        // pull back on the acceleration
+        modified_vel = (accel.normalized() * max_accel_cm) * dt + original_vel;
```

Comment:
We should clamp the value of `dt`.  It probably doesn't matter in that you'll just hit the other limit, but....

Reply:
I am not sure if `dt` requires an upperbound (and what that would be)? We sure do not check for it anywhere else in the avoidance code that we have since years.
But, I have added a bit more protection by including an early return if dt isn't positive (earlier I was just checking if it isn't zero).